src/EPR.Calculator.Frontend/Services/BillingInstructionsApiService.cs

```
@@ -0,0 +1,87 @@
+        /// </returns>
+        /// <remarks>
+        /// The method is not yet implemented and will throw a <see cref="NotImplementedException"/>.
+        /// </remarks>
+        public async Task<bool> PutAcceptRejectBillingInstructions(int calculationRunId, ProducerBillingInstructionsHttpPutRequestDto requestDto)
+        {
+            var apiUrl = this.GetApiUrl(ConfigSection.ProducerBillingInstructions, ConfigSection.ProducerBillingInstructionsV2HttpPut);
+
+            var response = await this.CallApi(HttpMethod.Put, apiUrl, calculationRunId.ToString(), requestDto);
+
+            return response.IsSuccessStatusCode;
+        }
+
+        protected async Task<HttpResponseMessage> CallApi(
+            HttpMethod httpMethod,
+            Uri apiUrl,
+            string argument,
+            ProducerBillingInstructionsHttpPutRequestDto dto)
+        {
+            var argsString = !string.IsNullOrEmpty(argument)
+                ? $"/{argument}"
+                : string.Empty;
+            argsString = !argument.Contains("&") ? argsString : $"?{argument}";
+            var contentString = JsonSerializer.Serialize(dto);
+            var request = new HttpRequestMessage(
+                httpMethod,
+                new Uri($"{apiUrl}{argsString}"));
+            request.Content = new StringContent(
+                contentString,
+                Encoding.UTF8,
+                StaticHelpers.MediaType);
+            var client = this.GetHttpClient(dto.AuthorizationToken);
+            return await client.SendAsync(request);
+        }
+
+        protected Uri GetApiUrl(string configSection, string configKey)
+            => new Uri(this.GetConfigSetting(configSection, configKey));
+
+        private HttpClient GetHttpClient(string authToken)
+        {
+            var client = clientFactory.CreateClient();
+            if (client.DefaultRequestHeaders is not null && !client.DefaultRequestHeaders.Contains("Authorization"))
```

When the check 
`client.DefaultRequestHeaders is not null`
fails,
Do you need to throw an error? Because this will anyway break further?

---

Done